### DISCUSSION TITLE
Prisma schema features

### DISCUSSION THREAD

**Discussion by leenorshn** [prisma/prisma#13211] [Q&A]
For example you are developping a small brank transfet system.
you have an **Account** model and **Transaction** model.

this is model Account:
```
model Account{
id String,
balance Number
owner String
}
```
 this is model Transaction's Structure
```
model Transaction{
id String
sender Account
receiver Account
amount Number
}
```
This is the logic that i want to impliment, but for now with **prisma** i have this error:

`Error validating model "Trans": Ambiguous relation detected. The fields `sender` and `receiver` in model `Trans` both refer to `User`. Please provide different relation names for them by adding `@relation(<name>`.

How can i work with this kind off model?

**Comment by nurul3101** [comment]
Hey @leenorshn 👋 

Here's the schema file which you could use:

```prisma
model Account {
  id                    String        @id @default(cuid())
  balance               Int           @db.Integer
  owner                 String
  transactionAsSender   Transaction[] @relation("TransactionAsSender")
  TransactionAsReceiver Transaction[] @relation("TransactionAsReceiver")
}

model Transaction {
  id                String  @id @default(cuid())
  sender            Account @relation(name: "TransactionAsSender", fields: [accountSenderId], references: [id])
  receiver          Account @relation(name: "TransactionAsReceiver", fields: [accountReceiverId], references: [id])
  amount            Int     @db.Integer
  accountSenderId   String
  accountReceiverId String
}
```

You are looking for disambiguating your relations: [reference](https://www.prisma.io/docs/concepts/components/introspection#disambiguating-relations).

**Reply by leenorshn** [reply]
Thanks a lot . We love all your works guys.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
